### ISSUE TITLE
"Required" rules are not being enforced at all by the model

### ISSUE THREAD

**Issue by khuderm** [codeigniter4/CodeIgniter4#2555]
Working on a very simple web app.
1. User Submits a form
2. I grab the data using getPOST and use fill to put the data into my DB entity.
3. I call insert but required fields are ignored and null or empty values are inserted into the db.

```
$postData = $this->request->getPOST();
$user = new App\Entities\User();
$userModel = new App\Models\UserModel();
// $user->fill($postData);
if (!$userModel->insert($user)) {
    return $userModel->errors();
}
```

In the user model, most of the fields are required but the above code will succeed no matter if I call fill(). The required fields are in the allowed fields array as well.

Ultimately, the issue seem to boil down to toRawArray call which is defined inside the Entities class. It seems to just return any user provided data as opposed to all the fields of the object filled with user provided data. 

Perhaps overriding the method in my entities class would solve this issue.

Basically, the required rule is outright ignored because the field does not exist in the dataset. This should cause a required error instead of attempting to insert. I can write a check in the controller or model to manually validate that single field but seems counter-intuitive to model validations.

**Comment by khuderm** [comment]
Perhaps overriding the method in my entities class would solve this issue.

Basically, the required rule is outright ignored because the field does not exist in the dataset. This should cause a required error instead of attempting to insert. I can write a check in the controller or model to manually validate that single field but seems counter-intuitive to model validations.

**Comment by khuderm** [comment]
Im just gonna go ahead and close this issue. I was going through the changelogs and docs again and noticed that the way fields are defined had changed. It is working as intended now. Thanks
